syntax = "proto3";
package huzch;
option cc_generic_services = true;

message UserInfo {
    string user_id = 1;
    string name = 2;
    string phone = 3;
    string description = 4;
    bytes  avatar = 5;
}

message ChatSessionInfo {
    optional string single_chat_friend_id = 1;
    string chat_session_id = 2;
    string chat_session_name = 3;
    optional MessageInfo prev_message = 4;
    optional bytes avatar = 5;
}

enum MessageType {
    STRING = 0;
    SPEECH = 1;
    IMAGE = 2;
    FILE = 3;
}
message StringMessageInfo {
    string content = 1;
}
message SpeechMessageInfo {
    optional string file_id = 1;
    optional bytes file_content = 2;
}
message ImageMessageInfo {
    optional string file_id = 1;
    optional bytes file_content = 2;
}
message FileMessageInfo {
    optional string file_id = 1;
    optional int64 file_size = 2;
    optional string file_name = 3;
    optional bytes file_content = 4;
}

message MessageContent {
    MessageType message_type = 1;
    oneof message_content {
        StringMessageInfo string_message = 2;
        SpeechMessageInfo speech_message = 3;
        ImageMessageInfo image_message = 4;
        FileMessageInfo file_message = 5;
    };
}

message MessageInfo {
    string message_id = 1;
    string chat_session_id = 2;
    int64 timestamp = 3;
    UserInfo sender = 4;
    MessageContent message = 5;
}

message FileDownloadData {
    string file_id = 1;
    bytes file_content = 2;
}

message FileUploadData {
    string file_name = 1;
    int64 file_size = 2;
    bytes file_content = 3;
}
